network/p2p/router_test.go

```diff
@@ -62,10 +67,10 @@ func TestAppRequestResponse(t *testing.T) {
 						return response, nil
 					})
 
-				callback := func(ctx context.Context, actualNodeID ids.NodeID, actualResponse []byte, err error) {
+				callback := func(ctx context.Context, actualNodeID ids.NodeID, actualResponse []byte, err *common.AppError) {
 					defer wg.Done()
 
-					require.NoError(t, err)
+					require.Nil(t, err)
```

Comment:
`NoError` doesn't seem to play well anymore now that we're using the concrete type instead of the `error` interface.

Reply:
I think we could go back using the require.NoError if we common.AppError had a value receiver instead of a pointer received and if we explicitly pass nil whenever the error is not there (vs a default error code signalling no error).
Not sure which alternative is cleaner.

Reply:
I think I understand what's happening now - we're passing `p2p.AppError` into the callback as error instead of an `error`, so this is a non-nil error with a nil pointer instead of a nil error. This is not correct according to go's error spec ([ref](https://go.dev/ref/spec#Errors)).

I'll update the parameter to be an `error` because it may or may not be nil depending if the request failed or not.